presto-geospatial-toolkit/src/main/java/com/facebook/presto/geospatial/GeometryUtils.java

```diff
@@ -42,4 +58,111 @@ public static boolean isEsriNaN(double d)
     {
         return Double.isNaN(d) || Double.isNaN(translateFromAVNaN(d));
     }
+
+    public static int getPointCount(OGCGeometry ogcGeometry)
+    {
+        GeometryCursor cursor = ogcGeometry.getEsriGeometryCursor();
+        int points = 0;
+        while (true) {
+            com.esri.core.geometry.Geometry geometry = cursor.next();
+            if (geometry == null) {
+                return points;
+            }
+
+            if (geometry.isEmpty()) {
+                continue;
+            }
+
+            if (geometry instanceof Point) {
+                points++;
+            }
+            else {
+                points += ((MultiVertexGeometry) geometry).getPointCount();
+            }
+        }
+    }
+
+    public static Envelope getEnvelope(OGCGeometry ogcGeometry)
+    {
+        GeometryCursor cursor = ogcGeometry.getEsriGeometryCursor();
+        Envelope overallEnvelope = new Envelope();
+        while (true) {
+            Geometry geometry = cursor.next();
+            if (geometry == null) {
+                return overallEnvelope;
+            }
+
+            Envelope envelope = new Envelope();
+            geometry.queryEnvelope(envelope);
+            overallEnvelope.merge(envelope);
+        }
+    }
+
+    public static boolean disjoint(Envelope envelope, OGCGeometry ogcGeometry)
+    {
+        GeometryCursor cursor = ogcGeometry.getEsriGeometryCursor();
+        while (true) {
+            Geometry geometry = cursor.next();
+            if (geometry == null) {
+                return true;
+            }
+
+            if (!GeometryEngine.disjoint(geometry, envelope, null)) {
+                return false;
+            }
+        }
+    }
+
+    public static boolean contains(OGCGeometry ogcGeometry, Envelope envelope)
+    {
+        GeometryCursor cursor = ogcGeometry.getEsriGeometryCursor();
+        while (true) {
+            Geometry geometry = cursor.next();
+            if (geometry == null) {
+                return false;
+            }
+
+            if (GeometryEngine.contains(geometry, envelope, null)) {
+                return true;
+            }
+        }
+    }
+
+    /**
+     * @return true if the geometry is a point or a rectangle
```

Comment:
![Image of Yaktocat](https://media1.tenor.com/images/e5e91360089456411ef660428fb48270/tenor.gif?itemid=9900289)